ethereum/rpc/apis.go

```diff
@@ -33,53 +32,80 @@ const (
 )
 
 // GetRPCAPIs returns the list of all APIs
-func GetRPCAPIs(ctx *server.Context, clientCtx client.Context, tmWSClient *rpcclient.WSClient) []rpc.API {
+func GetRPCAPIs(ctx *server.Context, clientCtx client.Context, tmWSClient *rpcclient.WSClient, selectedAPIs []string) []rpc.API {
 	nonceLock := new(types.AddrLocker)
-	backend := backend.NewEVMBackend(ctx.Logger, clientCtx)
-	ethAPI := eth.NewPublicAPI(ctx.Logger, clientCtx, backend, nonceLock)
+	evmBackend := backend.NewEVMBackend(ctx.Logger, clientCtx)
+	ethAPI := eth.NewPublicAPI(ctx.Logger, clientCtx, evmBackend, nonceLock)
 
-	return []rpc.API{
-		{
-			Namespace: Web3Namespace,
-			Version:   apiVersion,
-			Service:   web3.NewPublicAPI(),
-			Public:    true,
-		},
-		{
+	var apis []rpc.API
+
+	//Eth apis will be set by default
+	apis = append(apis,
+		rpc.API{
 			Namespace: EthNamespace,
 			Version:   apiVersion,
 			Service:   ethAPI,
 			Public:    true,
 		},
-		{
+	)
+	apis = append(apis,
+		rpc.API{
 			Namespace: EthNamespace,
 			Version:   apiVersion,
-			Service:   filters.NewPublicAPI(ctx.Logger, tmWSClient, backend),
-			Public:    true,
-		},
-		{
-			Namespace: NetNamespace,
-			Version:   apiVersion,
-			Service:   net.NewPublicAPI(clientCtx),
-			Public:    true,
-		},
-		{
-			Namespace: PersonalNamespace,
-			Version:   apiVersion,
-			Service:   personal.NewAPI(ctx.Logger, ethAPI),
-			Public:    true,
-		},
-		{
-			Namespace: TxPoolNamespace,
-			Version:   apiVersion,
-			Service:   txpool.NewPublicAPI(ctx.Logger),
-			Public:    true,
-		},
-		{
-			Namespace: DebugNamespace,
-			Version:   apiVersion,
-			Service:   debug.NewInternalAPI(ctx),
+			Service:   filters.NewPublicAPI(ctx.Logger, tmWSClient, evmBackend),
 			Public:    true,
 		},
+	)
+
+	for _, api := range selectedAPIs {
```

Comment:
Do we need to deduplicate here?

Reply:
the other problem I see is adding a namespace multiple times (eg `evm-rpc.apis=web3,web3,txpool,txpool`). We should either validate the slice or remove any duplicates

Reply:
And good catch 🙏 ! Fixed